scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -155,19 +159,35 @@ def _split_encodings(
         to_decode: list[bytes] = []
         while to_keep:
             encoding = to_keep.pop()
-            if encoding not in ACCEPTED_ENCODINGS:
+            if encoding not in supported_encodings:
```

Comment:
It seems it changes the behavior for x-gzip encoding; previously it wasn't removed from the headers, and probably wasn't decoded at all? Hm, does it mean that x-gzip handling code was not working before?

I think this change is for better, but we might want to mention it in the release notes.

Reply:
`x-gzip` handling was silently broken in 2.12.0 when this code was added.